### PR TITLE
Treat everything other than FxOS and desktop as android (bug 1180296)

### PR DESCRIPTION
**Pull request by mstriemer**
https://bugzilla.mozilla.org/show_bug.cgi?id=1180296

### CODE REVIEW

File: tests/capabilities.js [hunk 1]
```diff
@@ -117,8 +127,9 @@ define('tests/capabilities',
                 firefoxOS: false,
                 firefoxAndroid: false,
                 widescreen: function() { return false; },
+                os: {type: ''},
             };
-            assert.equal(capabilities.device_type(caps), 'desktop');
+            assert.equal(capabilities.device_type(caps), 'android-mobile');
         });
 
         it('is desktop on other tablet', function() {
```
Comment (diox) [inline tests/capabilities.js]: need to update this test name

File: tests/capabilities.js [hunk 2]
```diff
@@ -117,8 +127,9 @@ define('tests/capabilities',
                 firefoxOS: false,
                 firefoxAndroid: false,
                 widescreen: function() { return false; },
+                os: {type: ''},
             };
-            assert.equal(capabilities.device_type(caps), 'desktop');
+            assert.equal(capabilities.device_type(caps), 'android-mobile');
```
Comment (diox) [inline tests/capabilities.js]: need to update this test name (`'is desktop on other mobile'` is no longer true)

File: tests/capabilities.js [hunk 3]
```diff
@@ -108,6 +117,7 @@ define('tests/capabilities',
                 firefoxOS: false,
                 firefoxAndroid: true,
                 widescreen: function() { return true; },
+                os: {type: 'mobile'},
             };
             assert.equal(capabilities.device_type(caps), 'android-tablet');
```
Comment (diox) [inline tests/capabilities.js]: `os.type` is `mobile` and it ends up being an `android-tablet` ??

File: tests/capabilities.js (hunk 2 above)
Comment (diox) [inline tests/capabilities.js]: also, `os.type` is `''`, why is this considered `android-mobile` ?

File: tests/capabilities.js [hunk 4]
```diff
@@ -126,8 +137,9 @@ define('tests/capabilities',
                 firefoxOS: false,
                 firefoxAndroid: false,
                 widescreen: function() { return true; },
+                os: {type: ''},
```
Comment (diox) [inline tests/capabilities.js]: also, os.type is '', why is this considered android-tablet ?

File: core/capabilities.js [hunk 5]
```diff
@@ -101,23 +101,22 @@ define('core/capabilities', ['core/settings'], function(settings) {
         caps = caps || static_caps;
         if (caps.firefoxOS) {
             return 'firefoxos';
-        } else if (caps.firefoxAndroid) {
+        } else if (caps.os.type === 'desktop') {
+            return 'desktop';
+        } else {
             return 'android';
```
Comment (diox) [inline core/capabilities.js]: Add a comment explaining why we do this

**Comment by mstriemer** [comment]
Updated the comments. Do they make sense?

**Comment by diox** [comment]
Ok, wfm, r+

**Pull request merged**